src/Orchard.Web/Modules/Orchard.Azure/Services/FileSystems/AzureFileSystem.cs

```diff
@@ -239,7 +239,8 @@ public class AzureFileSystem {
                 }
 
                 if (blob is CloudBlobDirectory) {
-                    string foldername = blob.Uri.Segments.Last();
+                    var blobDir = (CloudBlobDirectory)blob;
+                    string foldername = blobDir.Prefix.Substring(blobDir.Parent.Prefix.Length);
```

Comment:
There is always at least tenant's folder, or "Default" folder without multitenancy.